### PR TITLE
improve tasks UI

### PR DESCRIPTION
**Pull request by Lonexera**
+ Tasks cards UI
+ No tasks UI
+ No plants UI

Screenshot_1641472744 Screenshot_1641472750

### CODE REVIEW

File: app/src/main/java/com/example/plantsapp/presentation/ui/utils/TaskExt.kt [hunk 1]
```diff
@@ -16,7 +16,16 @@ fun Task.getIconRes(): Int {
...
 }
+
+fun Task.getColorRes(): Int {
+    return when (this) {
+        is Task.WateringTask -> R.color.blue
+        is Task.SprayingTask -> R.color.purple
+        is Task.LooseningTask -> R.color.brown
+        is Task.TakingPhotoTask -> R.color.green_blue
+    }
+}
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/presentation/ui/utils/TaskExt.kt:31]: If it is used only once , maybe leave it as private inside of ViewHolder you use, if not - it is okay

File: app/src/main/res/layout/fragment_plants.xml [hunk 2]
```diff
@@ -34,6 +35,49 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (Vladuken) [inline app/src/main/res/layout/fragment_plants.xml:38]: Maybe move this to include?

File: app/src/main/res/layout/fragment_tasks.xml [hunk 3]
```diff
@@ -23,4 +24,48 @@
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@id/pb_loading" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (Vladuken) [inline app/src/main/res/layout/fragment_tasks.xml:28]: same here - how about move it to include?

**Review by Vladuken — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/plantsapp/presentation/ui/utils/TaskExt.kt (hunk 1 above)
Comment (Lonexera) [reply app/src/main/java/com/example/plantsapp/presentation/ui/utils/TaskExt.kt:31]: Planned to use this in next pull request on Plant Detail Screen UI improvement

File: app/src/main/java/com/example/plantsapp/presentation/ui/tasks/adapter/PlantWithTasksViewHolder.kt [hunk 4]
```diff
@@ -32,6 +39,9 @@ class PlantWithTasksViewHolder(
         }
     }
 
+    private val List<TaskWithState>.notCompletedAmount: Int
+        get() = filter { !it.isCompleted }.count()
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/presentation/ui/tasks/adapter/PlantWithTasksViewHolder.kt:43]: you can call count { !it.isCompleted } right here

File: app/src/main/java/com/example/plantsapp/presentation/ui/tasks/TasksFragment.kt [hunk 5]
```diff
@@ -80,9 +82,18 @@ class TasksFragment : Fragment(R.layout.fragment_tasks) {
                 layoutManager = LinearLayoutManager(context)
                 adapter = plantsWithTasksAdapter
             }
+
+            layoutNoTasks.ivNoItemsImage.setImageResource(R.drawable.ic_baseline_calendar_24)
+            layoutNoTasks.tvNoItemsTitle.setText(R.string.title_no_tasks)
+            layoutNoTasks.tvNoItemsDetail.setText(R.string.msg_no_tasks)
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/presentation/ui/tasks/TasksFragment.kt:88]: In such cases I write Extension function on ***Binding , to have one method that will change all needed data

**Review by Vladuken — APPROVED** [review]

**Pull request merged**